apps/wing/src/analytics/storage.ts

```diff
@@ -0,0 +1,94 @@
+import { readFileSync, writeFileSync, mkdirSync, existsSync } from "fs";
+import { AnalyticEvent } from "./event";
+import { randomBytes } from "crypto";
+import {v4 as uuidv4 } from "uuid";
+import path from "path";
+import * as os from "os";
+
+const WING_HOME_DIR = path.join(os.homedir(), ".wing")
+const WING_ANALYTICS_CONFIG_FILE = path.join(WING_HOME_DIR, 'wing-analytics-config.json');
+
+interface AnalyticsConfig {
+  anonymousId: string;
+  disclaimerDisplayed?: boolean;
+  optOut?: boolean;
+}
+
+/**
+ * Stores an analytic event to disk for later export
+ * 
+ * @param event the event to store
+ * @returns string the file path of the stored analytic
+ */
+export function storeAnalyticEvent(event: AnalyticEvent): string {
+  const eventId = uuidv4();
+  const tmpdir = path.join(os.tmpdir(), "wing-analytics");
+  
+  if (!existsSync(tmpdir)) {
+    mkdirSync(tmpdir);
+  }
+
+  const analyticReportFile = path.join(tmpdir, `${eventId}.json`);
+  
+  const anonymousId = getAnonymousId();
+
+  // attach timestamp and anonymousId to event
+  event.timestamp = event.timestamp ?? new Date().toISOString();
+  event.anonymousId = anonymousId;
+
+  saveAnalyticsReport(analyticReportFile, event);
+
+  return analyticReportFile;
+}
+
+export function getAnonymousId(): string {
+  let config = loadAnalyticsConfig();
+  return config.anonymousId;
+}
+
+export function loadAnalyticsConfig(): AnalyticsConfig {
+  try {
+    const fileContents = readFileSync(WING_ANALYTICS_CONFIG_FILE, 'utf-8');
+    return JSON.parse(fileContents) as AnalyticsConfig;
+  } catch (error) {
+    // Possible issue reading analytics config file, first determine if file exists yet
+    if (existsSync(WING_ANALYTICS_CONFIG_FILE)) {
```

Comment:
nit: Instead of checking for existence, check out the error codes from the error to see if it's ENOENT or EACCES.